packages/editor/src/toolbar/popups/color-picker.tsx

```diff
@@ -51,23 +52,30 @@ type ColorPickerProps = {
   onChange: (color: string) => void;
   onClose?: () => void;
   title?: string;
+  onSave?: () => void;
+  cacheKey?: string;
 };
 const PALETTE_SIZE = [35, 35, 25];
 export function ColorPicker(props: ColorPickerProps) {
   const {
-    colors = DEFAULT_COLORS,
+    colors = [
+      ...DEFAULT_COLORS,
+      ...(config.get<string[]>(props.cacheKey!, []) || [])
+    ],
```

Comment:
`colors` is a prop which means that it can be used by each instance to specify it's own set of custom colors.

The current approach uses the `cacheKey` to get the custom colors on every render. Aside from being inefficient, it also makes it super hard to update the custom colors when a color is added/deleted.

Here's what we should do instead:
1. Take the custom colors as a prop
2. Create a new state for storing all the colors (default & custom).
3. Use a `useEffect` hook to `push` the custom colors into the state.